packages/react-instantsearch-hooks/src/lib/__tests__/useSearchResults.test.tsx

```diff
@@ -49,4 +53,47 @@ describe('useSearchResults', () => {
     });
     expect(result.current.results.__isArtificial).toBeUndefined();
   });
+
+  test('should not return null results when first search is stalled', async () => {
+    const wrapper = createInstantSearchTestWrapper({
+      stalledSearchDelay: 0,
+      searchClient: createSearchClient({
+        search: async () => {
+          // Simulate a stalled search
+          await new Promise((resolve) => setTimeout(resolve, 50));
+
+          return createMultiSearchResponse();
+        },
+      }),
+    });
+    const { result, waitForNextUpdate } = renderHook(() => useSearchResults(), {
+      wrapper: ({ children }) =>
+        wrapper({
+          children: (
+            <>
+              <SearchBox />
+              {children}
+            </>
+          ),
+        }),
+    });
+
+    // Initial render
+    expect(result.current).toEqual({
+      results: expect.any(SearchResults),
+      scopedResults: [
+        expect.objectContaining({
+          helper: expect.any(AlgoliaSearchHelper),
+          indexId: 'indexName',
+          results: expect.any(SearchResults),
+        }),
+      ],
+    });
+    expect(result.current.results.__isArtificial).toEqual(true);
```

Comment:
This is covered in the first test already and unrelated to the test scenario, I don't think we need it twice even if the parameters are slightly different.